index.js

```diff
@@ -24,48 +24,40 @@ SOFTWARE.
 
 'use strict';
 
-function skip(input, skipIndex) {
-    return input.filter(function(item, index) {
-        return Boolean(index > skipIndex - 1);
-    });
-}
+function between(arr, start, end) {
+    validateArguments(arr, start, end);
 
-function limit(input, limitIndex) {
-    return input.filter(function (item, index) {
-        return Boolean(index <= limitIndex);
-    });
-}
+    const skipped = skip(arr, start);
 
-function throwTypeError(message) {
-    throw new TypeError(message);
+    return limit(skipped, end - start);
 }
 
+
 function validateArguments(arr, start, end) {
-     if (!Array.isArray(arr)) {
-        throwTypeError(`Expected an Array, got ${typeof arr}`);
+    if (!Array.isArray(arr)) {
+        new TypeError(`Expected an Array, got ${typeof arr}`);
     }
 
     if (typeof start !== 'number') {
-        throwTypeError(`Expcted start index to be number, got ${typeof start}`);
+        new TypeError(`Expcted start index to be number, got ${typeof start}`);
     }
 
-    if(typeof end !== 'number') {
-        throwTypeError(`Expected end index to be number, got ${typeof end}`);
+    if (typeof end !== 'number') {
+        new TypeError(`Expected end index to be number, got ${typeof end}`);
     }
 
     if (end < start) {
-        throw new RangeError('Invalid range provided');
+        throw new RangeError(`Invalid range provided, start index (${start}) is bigger than end index (${end})`);
     }
 }
 
-function between(arr, start, end) {
-    validateArguments(arr, start, end);
-
-    const skipped = skip(arr, start);
-
-    return limit(skipped, end - start);
+function skip(input, skipIndex) {
+  return input.filter(item, index) => index > skipIndex - 1;
 }
 
+function limit(input, limitIndex) {
+    return input.filter(item, index) => index <= limitIndex;
```

Comment:
Malformed arrow function
```suggestion
    return input.filter((item, index) => index <= limitIndex);
```